lib/inspec/base_cli.rb

```diff
@@ -80,74 +80,97 @@ def self.exec_options
     def self.default_options
       {
         exec: {
-          'reporter' => {
-            'cli' => nil,
-          },
+          'reporter' => ['cli'],
           'show_progress' => false,
           'color' => true,
           'create_lockfile' => true,
           'backend_cache' => false,
         },
+        shell: {
+          'reporter' => ['cli'],
+        },
       }
     end
 
-    def self.clean_reporters(opts)
-      # Parse report options from CLI
+    def self.parse_reporters(opts)
+      # merge in any legacy formats as reporter
+      # this method will only be used for ad-hoc runners
+      if !opts['format'].nil? && opts['reporter'].nil?
+        warn '[DEPRECATED] The option --format is being is being deprecated and will be removed in inspec 3.0. Please use --reporter'
+        opts['reporter'] = [opts['format']]
+        opts.delete('format')
+      end
+
+      # parse out cli to proper report format
       if opts['reporter'].is_a?(Array)
-        stdout = 0
         reports = {}
         opts['reporter'].each do |report|
           k, v = report.split(':')
-          v = nil if v == '-'
-          reports[k] = v
-          stdout += 1 if v.nil?
+          if v.nil? || v.strip == '-'
+            reports[k] = { 'stdout' => true }
+          else
+            reports[k] = {
+              'file' => v,
+              'stdout' => false,
+            }
+          end
         end
-
-        raise ArgumentError, 'The option --reporter can only have a single report outputting to stdout.' if stdout > 1
-
         opts['reporter'] = reports
       end
 
-      # Catch anyone using the legacy --format option
-      unless opts['format'].nil?
-        opts['reporter'] = { opts['format'] => nil }
-        opts.delete('format')
+      # add in stdout if not specified
+      if opts['reporter'].is_a?(Hash)
+        opts['reporter'].each do |k, v|
+          opts['reporter'][k] = {} if v.nil?
+          opts['reporter'][k]['stdout'] = true if opts['reporter'][k].empty?
+        end
       end
 
-      # Catch any dynamic inspec runners or tests
-      opts['reporter'] = { 'cli' => nil } if opts['reporter'].nil?
-
+      validate_reporters(opts['reporter'])
       opts
     end
 
+    def self.validate_reporters(reporters)
+      return if reporters.nil?
+
+      valid_types = [
+        'json',
+        'json-min',
+        'json-rspec',
+        'cli',
+        'junit',
+        'html',
+        'documentation',
+        'progress',
+      ]
+
+      reporters.each do |k, _v|
+        raise NotImplementedError, "'#{k}' is not a valid reporter type." unless valid_types.include?(k)
+      end
+
+      # check to make sure we are only reporting one type to stdout
+      stdout = 0
+      reporters.each_value do |v|
+        stdout += 1 if v&.[]('stdout')&.==(true)
```

Comment:
heh, gotta love the safe navigation operator! But this syntax is super-hard to grok, especially the `[]` usage here.

In this case, I'd recommend something a bit more standard/legacy for readability reasons:

```ruby
stdout += 1 if v.key?('stdout') && v['stdout'] == true
```

... or, honestly, if you're validating the reporter data structure before you even get to this point, you can probably just do:

```ruby
stdout += 1 if v['stdout'] == true
```